pages/api/webhooks/posts.ts

```diff
@@ -28,14 +29,19 @@ export default async function handler(
     return res.status(401).json({ error: 'Unauthorized' });
   }
 
-  if (!req.body.record) {
+  if (!req.body.record && !req.body.old_record) {
     return res.status(400).json({ error: 'No record provided' });
   }
 
   const {
-    record: { title, short, slug, published_at, user, language },
+    record: { title, short, slug, published, published_at, user, language },
+    old_record: { published: old_published } = { published: false },
   } = req.body;
 
+  if (!published || old_published) {
+    return res.status(400).json({ error: 'Post not published yet' });
+  }
```

Comment:
I am not sure if it's good to return an error code to the supabase webhook.
Not sure, what happens, when it things it was failing.

If "published" is not changed to `true`, we could `200` (because it's a valid request).

```suggestion
  if (!published || old_published) {
    return res.status(200).json({ message: 'Post was not published' });
  }
```